docs: update test results in POSIX compatibility (#7125)
juicedata/juicefs: docs/en/reference/posix_compatibility.md

### Test environment

- Host: Aliyun ESC server
- OS: Ubuntu 24.04 (Kernel `6.8.0-55-generic`)
- JuiceFS version: 1.4-beta2
- LTP version: 20260529

### Test steps

1. Download the LTP [release](https://github.com/linux-test-project/ltp/releases/download/20260529/ltp-full-20260529.tar.xz) from GitHub.

2. Unarchive, compile, and install LTP:

   ```bash
   tar -xvf ltp-full-20260529.tar.xz
   cd ltp-full-20260529
   ./configure
   make all
   make install
   ```

3. Install kirk (the new LTP test runner that replaces the removed runltp):

   ```bash
   pip install kirk
   ```

4. Change the directory to `/opt/ltp` where the test tools are installed:

   ```bash
   cd /opt/ltp
   ```

> 5. Create test configuration files. To simplify testing, delete pressure tests, filesystem-unrelated entries, and tests not run on JuiceFS from `runtest/fs` and `runtest/syscalls`, modified files are saved as `fs-jfs` and `syscalls-jfs` (see [Appendix](#appendix) for the detailed deletion list):

Reviewer: ```suggestion
5. Create test configuration files. To simplify testing, delete pressure tests, filesystem‑unrelated entries, and tests that do not run on JuiceFS from `runtest/fs` and `runtest/syscalls`. Then, save the modified files as `fs‑jfs` and `syscalls‑jfs`. (See [Appendix](#appendix) for the detailed deletion list.)
```

6. Execute tests:

   ```bash
   export LTPROOT=/opt/ltp
   export PATH=/opt/ltp/testcases/bin:$PATH
   cd /mnt/jfs
   kirk --run-suite fs_bind --sut default --tmp-dir /mnt/jfs
   kirk --run-suite fs_perms_simple --sut default --tmp-dir /mnt/jfs
   kirk --run-suite smoketest --sut default --tmp-dir /mnt/jfs
   kirk --run-suite fs-jfs --sut default --tmp-dir /mnt/jfs
   kirk --run-suite syscalls-jfs --sut default --tmp-dir /mnt/jfs
   kirk --run-suite fcntl-locktests --sut default --tmp-dir /mnt/jfs
   ```

### Test result

> | Test suite | Runs | Passed | Failed | Broken | Skipped |

Reviewer: ```suggestion
| Test suite | Tests run | Passed | Failed | Broken | Skipped |
```

|------------|------|--------|--------|--------|---------|
| fs_bind | 95 | 95 | 0 | 0 | 0 |
| fs_perms_simple | 18 | 18 | 0 | 0 | 0 |
| smoketest | 12 | 10 | 0 | 0 | 2 |
| fs-jfs | 30 | 30 | 0 | 0 | 0 |
| syscalls-jfs | 1323 | 1300 | 0 | 0 | 23 |
| fcntl-locktests | 1 | 1 | 0 | 0 | 0 |
| **Total** | **1479** | **1454** | **0** | **0** | **25** |

> All file system related tests run on JuiceFS passed with 0 failures and 0 broken.

Reviewer: ```suggestion
All file system-related tests run on JuiceFS passed, with 0 failures and 0 broken tests.
```

quota_remount_test01 quota_remount_test01.sh
isofs isofs.sh

## syscalls --> syscalls-jfs

> # Only key items related to FUSE and JuiceFS incompatibility are listed here.

Reviewer: ```suggestion
# A large number of unrelated test cases were removed from the syscalls suite. Only some cases unsupported by FUSE or JuiceFS are listed here.
```

# For the complete list, please refer to .github/workflows/bash/rm_syscalls

> # Pure kernel functionality tests, unrelated to userspace filesystems

Reviewer: ```suggestion
# Pure kernel functionality tests, unrelated to userspace filesystems.
```

close_range01 close_range01
close_range02 close_range02
openat201 openat201
openat202 openat202
openat203 openat203

# CVE regression tests (SELINUX), unrelated to JuiceFS
listxattr04 listxattr04
io_uring02 io_uring02

> # Tests skipped due to environment issues, require newer kernel versions

Reviewer: ```suggestion
# Tests skipped due to environment limitations; they require a newer kernel version
```

listmount04 listmount04
name_to_handle_at03 name_to_handle_at03
statmount09 statmount09
mount08 mount08

# Deprecated system calls, now replaced by getdents64
readdir21 readdir21

# Tests explicitly skipping FUSE
file_attr01 file_attr01
file_attr02 file_attr02
file_attr03 file_attr03
file_attr04 file_attr04
file_attr05 file_attr05
ioctl_fiemap01 ioctl_fiemap01
openat02 openat02
unlink09 unlink09
mount03 mount03

> # FUSE kernel module unsupported tests

Reviewer: ```suggestion
# Tests unsupported by the FUSE kernel module
```

fanotify10 fanotify10
fanotify13 fanotify13
fanotify16 fanotify16
fanotify18 fanotify18
fanotify19 fanotify19

> # JuiceFS does not support FS_NOCOW_FL flag

Reviewer: ```suggestion
# JuiceFS does not support the FS_NOCOW_FL flag
```
